test_util/EclRegressionTest.cpp

```diff
@@ -902,7 +902,11 @@ void ECLRegressionTest::results_smry()
                     keywords1[i].substr(0,5) == "WWPR:" ||
                     keywords1[i].substr(0,5) == "WGPR:" ||
                     keywords1[i].substr(0,5) == "WBHP:" ||
-                    keywords1[i].substr(0,4) == "SPR:")  {
+                    keywords1[i].substr(0,4) == "SPR:"  ||
+                    keywords1[i].substr(0,4) == "SPRD:" ||
+                    keywords1[i].substr(0,4) == "SPRDH:"||
+                    keywords1[i].substr(0,4) == "SPRDF:"||
+                    keywords1[i].substr(0,5) == "SPRDA:" ) {
```

Comment:
The lengths here aren't quite correct&mdash;e.g., line 909 should be `substr(0,6)`&mdash;but this only illustrates a deeper issue.  Maintaining such lists is error prone.  I think I'd prefer a slightly more abstract solution that puts the keyword list into a function and uses that.  E.g., something along the lines of
```C++
bool isIntegrationKw(const std::string& kw)
{
    static const auto kwlist = std::vector<std::string> {
        "WOPR", "WWPR", "WGPR", "WBHP",
        "SPR", "SPRD", "SPRDH", "SPRDF", "SPRDA",
    };

    const auto p = kw.find_first_of(':');
    return std::find(kwlist.begin(), kwlist.end(), kw.substr(0, p)) != kwlist.end();
}
```
Then the the `keywords1` loop becomes roughly
```C++
for (const auto& kw : keywords1) {
    if (! isIntegrationKw(kw)) { continue; }
   
    auto search2 = std::find(keywords2.begin(), keywords2.end(), kw);
    if (search2 != keywords2.end()) {
        keywords.push_back(kw);
    }
}
```

Reply:
Thanks for the comments. I had some impression that I ran to this (length of the keywords) before, not sure why it was mistaken again. Anyway, I will follow your suggestion.